src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -114,20 +126,49 @@ static bool analyze(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> f
                 return !failedList.Any();
             }
 
+            static bool analyzeSymbols(ElfBinary binary, out List<SymbolEntry<ulong>> failedList)
+            {
+                failedList = new List<SymbolEntry<ulong>>();
+
+                var symbols = binary.ELF.Sections.FirstOrDefault(s => s.Type == SectionType.DynamicSymbolTable) as SymbolTable<ulong>;
+                foreach (SymbolEntry<ulong> symbol in symbols.Entries)
+                {
+                    if (symbol.Name != "__stack_chk_fail" && symbol.Name != "stack_chk_guard" && symbol.Name != "__intel_security_cookie")
+                    {
+                        failedList.Add(symbol);
+                    }
+                    else
+                    {
+                        return true;
```

Comment:
>return [](http://example.com/codeflow?start=24&length=6)

OK, so I *think* what we're trying to do here is return true if we find any evidence of a stack protector function, yes? If so, fair enough. But why do we need to maintain a list of symbols that don't comprise a stack protector function? We wouldn't report on this,  right? 

If I'm correct, you can remove all references to the failed list and simply return false if we successfully exit the foreach loop.

Reply:
Ah ok. Yeah I don't think we really need that list of symbols. I'll remove it.